### PR TITLE
Audio Recording Interface

### PR DESCRIPTION
**Pull request by jjhuang1**

### CODE REVIEW

File: src/main/kotlin/device/IAudioRecorder.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package device
+
+import io.reactivex.Observable
+
+interface IAudioRecorder {
+    fun record()
+    fun stop()
+    fun getAudioByteObservable(): Observable<ByteArray>
```
Comment (jsarabia) [inline src/main/kotlin/device/IAudioRecorder.kt]: how about getAudioStream()?

Comment (jjhuang1) [reply src/main/kotlin/device/IAudioRecorder.kt]: Sure I'll change that

File: src/main/kotlin/device/IAudioRecorder.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package device
+
+import io.reactivex.Observable
+
+interface IAudioRecorder {
+    fun record()
```
Comment (jsarabia) [inline src/main/kotlin/device/IAudioRecorder.kt]: how about start()

Comment (jsarabia) [reply src/main/kotlin/device/IAudioRecorder.kt]: also; does AudioClip have a "pause?"

Comment (jjhuang1) [reply src/main/kotlin/device/IAudioRecorder.kt]: Not currently. Is that something we should add?

**Review by jsarabia — APPROVED** [review]

**Pull request merged**